src/utils/representation.py

```
@@ -35,11 +39,71 @@ class PromptRepresentation(BaseModel):
+    implicit: list[ImplicitObservationBase] = Field(
+        description="Facts CLEARLY IMPLIED by the user's message - atomic propositions derived through obvious implication. Example: ['Maria attended college' (from 'I graduated from college')]",
+        default_factory=list,
+    )
     deductive: list[DeductiveObservationBase] = Field(
         description="Conclusions that MUST be true given explicit facts and premises - strict logical necessities. Each deduction should have premises and a single conclusion.",
         default_factory=list,
     )
 
+    @field_validator('explicit', mode='before')
+    @classmethod
+    def normalize_explicit(cls, v):
```

Ruff (0.14.2)

Missing return type annotation for classmethod `normalize_explicit`

(ANN206)